samples/logging-sample/Dockerfile

```diff
@@ -14,9 +14,9 @@
 
 # [START cloudbuild_logging_sample_dockerfile]
 
-FROM golang AS build-env
+FROM golang:1.15 AS build-env
```

Comment:
Any particular reason we want 1.15? I haven't really thought about what we default to otherwise 😅

Reply:
Reading your comment now - any reason 1.16 didn't work for you? IIRC, there were some issues with the Go builder for GCB recently (that should be resolved now).

Reply:
Well 1.16 is the latest one (as far as I can tell!) and it failed to build properly -- had some errors regarding indirect requirements.  Probably should fix this sample to work with 1.16, but this was the easier solution.  WDYT?

Reply:
Fixed!  I guess the previous version of golang imported implicit requirements, whereas 1.16 didn't.  This was fixed by running `go mod tidy`